test/fail_compilation/fail12901.d

```diff
@@ -1,14 +0,0 @@
-/*
-TEST_OUTPUT:
----
-fail_compilation/fail12901.d(11): Error: constructor fail12901.S.this in and out contracts require function body
----
-*/
-
-struct S
-{
-    int a;
-    this(int n)
-    in { a = n; }
-    // no body
-}
```

Comment:
What's the motivation for removing this? Shouldn't it still be forbidden?

Reply:
Not sure as this is would be the third person to rebase and submit this patch.

It certainly seems reasonable to me to not include body-less ctors, dtors, and postblits to have contracts in the relaxation of this rule.

Reply:
Hmm according to the comment:

> Contracts can only appear without a body when they are virtual interface functions or abstract

and its respective if condition, this should still be forbidden.
I just checked locally and it doesn't throw an error anymore.
The problem is this jump:

```d
if (StructDeclaration sd = parent.isStructDeclaration())
{
    if (funcdecl.isCtorDeclaration())
    {
        goto Ldone;
     }
}
```

Reply:
Ah, interesting. I wonder the reason for that.